src/enzyme_ad/jax/Passes/LibDeviceFuncsRaisingPass.cpp

```diff
@@ -179,6 +184,117 @@ class CallToOpRaising : public OpRewritePattern<LLVM::CallOp> {
   StringAttr funcName;
 };
 
+class EnzymeAutodiffOpRaising : public OpRewritePattern<LLVM::CallOp> {
+public:
+  using OpRewritePattern<LLVM::CallOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(LLVM::CallOp op,
+                                PatternRewriter &rewriter) const override {
+    CallInterfaceCallable callable = op.getCallableForCallee();
+    auto callee = dyn_cast<SymbolRefAttr>(callable);
+    if (!callee)
+      return failure();
+    if (!callee.getLeafReference().strref().contains("__enzyme_autodiff"))
+      return failure();
+    Operation::operand_range operands = op.getArgOperands();
+    auto targetAddr =
+        dyn_cast<LLVM::AddressOfOp>(operands.front().getDefiningOp());
+    if (!targetAddr) {
+      op.emitError() << "first operand of __enzyme_autodiff was not an "
+                        "llvm.mlir.addressof op";
+      return failure();
+    }
+
+    FlatSymbolRefAttr funcToDiffSymbol = targetAddr.getGlobalNameAttr();
+    auto mod = op->getParentOfType<ModuleOp>();
+    auto funcToDiff =
+        cast<FunctionOpInterface>(mod.lookupSymbol(funcToDiffSymbol));
+
+    // Infer the argument activities
+    SmallVector<Value> arguments;
```

Comment:
mind moving this to a helper function (that way it can be reused for fwddiff op raising)